driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -228,6 +286,18 @@ public static DocumentExpression of(final Bson document) {
                 document.toBsonDocument(BsonDocument.class, cr))));
     }
 
+    /**
+     * The null value in the context of the MongoDB Query Language (MQL).
+     *
+     * <p>The null value is not part of, and cannot be used as if it were part of,
+     * any other type. It has no explicit type of its own. Instead of checking for
```

Comment:
"cannot be used as if it were part of any other type" -> "cannot be used as if it were part of any other **explicit** type"?

Reply:
What non-explicit other (not-null) type can it be used as part of?

Reply:
[marked as optional]

It's own type that has a single null value. That's what Java does for `null`, and that's the type one may infer (an implicit type) for null values for MQL similarly how this API inferred numbers and integers.

This API does not allow using the null value as if it were of any ("part of") type explicitly defined by the API, or were convertible to any explicit type. But like any other value, the null value has a type (if not explicit, then implicit).

Reply:
I tried to clarify a bit.